### PR TITLE
[docs] Update index and quicktour

### PR DESCRIPTION
**Pull request by stevhliu**
Starts implementing some of the things we discussed in our docs plan:

- clarify the difference between using `AutoPeftModel` (inference) and `PeftConfig` + `get_peft_model` (train) and splits the quicktour more generally along these lines
- add API reference for `AutoPeftModel` classes
- cleans up the index with a Space for discovering what models/methods are supported given a task (let me know if you think this is easier and cleaner)
- create separate subsection for LoRA guides

### CODE REVIEW

File: docs/source/_toctree.yml
Comment (BenjaminBossan) [inline docs/source/_toctree.yml:18]: I like this structure, nice.

Strangely, in the doc preview, this section shows as all capitalized "LORA" for me, not sure why. Not a big deal, just curious.

File: docs/source/index.md
Comment (BenjaminBossan) [inline docs/source/index.md:48]: I have no idea about iframes, but can the height be dynamic, so that we don't have to scroll for longer lists?

**Review by BenjaminBossan — COMMENTED** [review]
Thanks, Steven, this looks like it will help give a better structure to the PEFT docs and make the intro easier to understand for new users.

I left  a couple of comments, please take a look.

File: docs/source/quicktour.md
Comment (stevhliu) [reply docs/source/quicktour.md]: I think this can be something we put in the Tutorial section (this is good basic info for new users to learn since `AutoPeftModel` is just a wrapper around `PeftModel`)? For the quicktour though, I think we should aim for convenience to get users started as quickly as possible :)

File: docs/source/index.md
Comment (stevhliu) [reply docs/source/index.md:45]: Currently, the underlying [dataset](https://huggingface.co/datasets/stevhliu/supported-peft-models-methods) would need to be updated each time a new model is added or if a new method is supported. Let me know if you have any suggestions for automating it though, that'd be nice!

Comment (BenjaminBossan) [reply docs/source/index.md:45]: Technically, it would be possible. For example, we could set up a cron job or GitHub CI to parse the README (assuming that should be the single source of truth) with pandoc, extract the table, and create an update the on the dataset. But that seems like overkill for now. Probably we, the maintainers, just need to remember to update the dataset each time a new model is supported.

**Review by BenjaminBossan — APPROVED** [review]
Thanks for improving the docs, LGTM. I think it would be good to have another approval by Sourab or Younes before merging.

File: docs/source/quicktour.md
Comment (younesbelkada) [inline docs/source/quicktour.md:147]:
```suggestion
outputs = model.generate(input_ids=inputs["input_ids"].to("cuda"), max_new_tokens=50)
print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0])
```
The `torch.no_grad()` context manager is already used in `generate` AFAIK

**Pull request merged**
